sessiontxn/staleread/processor.go

```diff
@@ -206,7 +239,7 @@ func (p *staleReadProcessor) evaluateFromTxn() error {
 			nil,
 		)
 	}
-	return p.setAsNonStaleRead()
+	return p.setAsNonExplicitStaleRead()
```

Comment:
Does it mean that when we do this:

```
begin
select * from t;
set @@tidb_enable_external_ts_read=1;
select * from t;
```

The two select may return different values?

Reply:
Oops. I think it's not expected, as the `tidb_external_ts` could promote while executing the transaction. I'll fix this problem.

Reply:
Fixed. Also add a test for it.